### PR TITLE
V5 cluster tests new ones and fixes

### PR DESCRIPTION
**Pull request by ograu**
Tests rewritten with `nock`.

### CODE REVIEW

**Comment by ograu** [comment]
This is a WIP still! 
Tests had a different output in CircleCI.
Sorry for the inconvenience.

**Comment by oponder** [comment]
No worries. Let me know when you want a review!

**Comment by ograu** [comment]
Here I included fixes for node pool creation and for the login test issue I was experiencing in CircleCI, cause I needed both to complete the tests. So in order to exclude these changes from the diffs, we would have to merge first #835 and https://github.com/giantswarm/happa/pull/837

Ready for review once these are merged.

Tests included here: https://github.com/giantswarm/happa/issues/759 

Key pairs and app tabs not included in this PR and I am not planning to do these tests now.

**Comment by ograu** [comment]
cc/ @oponder this is ready for review

File: __tests__/V5ClusterManagement.js [hunk 1]
```diff
@@ -0,0 +1,458 @@
...
+  expect(modalTitle).toHaveTextContent(nodePool.id);
+
+  const inputMin = getByLabelText(/minimum/i);
+  const inputMax = getByLabelText(/maximum/i);
+
+  // Are the correct values in the correct fields?
+  expect(inputMin.value).toBe(defaultScaling.min.toString());
+  expect(inputMax.value).toBe(defaultScaling.max.toString());
+
+  // Change the values and modify the scaling settingsw
```
Comment (oponder) [inline __tests__/V5ClusterManagement.js]: typo: settingsw -> settings

File: __tests__/V5ClusterManagement.js [hunk 2]
```diff
@@ -0,0 +1,458 @@
...
+  const textButton = `Increase minimum number of nodes by ${newScaling.min}`;
+
+  await wait(() => getByText(textButton));
+
+  const submitButton = getByText(textButton);
+  fireEvent.click(submitButton);
+
+  //Wait for the Flash message to appear
+  await wait(() => {
+    getByText(/succesfully edited node pool name/i);
```
Comment (oponder) [inline __tests__/V5ClusterManagement.js]: We're scaling here, but expecting a message that confirms a name change?

Comment (oponder) [reply __tests__/V5ClusterManagement.js]: Indicates maybe something wrong in the code, not in the test.

File: __tests__/V5ClusterManagement.js [hunk 3]
```diff
@@ -0,0 +1,458 @@
...
+    debug,
+    container,
+  } = renderRouteWithStore(ROUTE, div, {});
+
+  // Wait for the view to render
+  await wait(() => {
+    expect(getByText(V5_CLUSTER.name)).toBeInTheDocument();
+  });
+
+  // fireEvent.click(getAllByText('•••')[0]);
```
Comment (oponder) [inline __tests__/V5ClusterManagement.js]: Let's not commit commented out code.

File: __tests__/V5ClusterManagement.js [hunk 4]
```diff
@@ -0,0 +1,458 @@
...
+
+  // Expect the cluster is not in the clusters list.
+  await wait(() => {
+    expect(queryByTestId(cluster.id)).not.toBeInTheDocument();
+  });
+
+  // This is not inside the component tree we are testing and so it is not cleaned up
+  // after test, so we have to remnove it manually in order to not cause conflicts with
+  // the next test with a flash message
+  document.querySelector('#noty_layout__topRight').remove();
```
Comment (oponder) [inline __tests__/V5ClusterManagement.js]: This feels a bit weird to me. The tests share a "document " somehow? I didn't know that.

File: __tests__/V5ClusterManagement.js (hunk 2 above)
Comment (ograu) [reply __tests__/V5ClusterManagement.js]: These flash messages are outside of the rendered `div` so they do not dissapear when component is unmounted after each test. What is happening here is that we are finding the flash message that triggered the test before.

So the only thing that it is wrong is that we are not expecting the right message.

I have tried to remove the flash message in the end of the tests when they are interfering with other tests.

File: __tests__/V5ClusterManagement.js (hunk 4 above)
Comment (ograu) [reply __tests__/V5ClusterManagement.js]: Yeah we are testing with a simulator of athe browser: `jsdom`

This is regarding what I wrote above: https://github.com/giantswarm/happa/pull/832#discussion_r347935946

**Comment by ograu** [comment]
thanks @oponder ! Fixing all the things and merging

File: __tests__/V5ClusterManagement.js (hunk 4 above)
Comment (ograu) [reply __tests__/V5ClusterManagement.js]: yep,  `jest` uses `jsdom` wich is a browser simulator and produces a document object

**Pull request merged**